paasta_tools/api/views/autoscaler.py

```diff
@@ -45,6 +51,36 @@ def update_autoscaler_count(request):
     service = request.swagger_data.get('service')
     instance = request.swagger_data.get('instance')
     desired_instances = request.swagger_data.get('json_body')['desired_instances']
+
+    try:
+        service_config = load_marathon_service_config(
+            service=service,
+            instance=instance,
+            cluster=settings.cluster,
+            soa_dir=settings.soa_dir,
+            load_deployments=False,
+        )
+    except:
+        error_message = 'Unable to load service config for %s.%s' % (service, instance)
+        raise ApiFailure(error_message, 404)
+
+    max_instances = service_config.get_max_instances()
+    if max_instances is None:
+        error_message = 'Autoscaling is not enabled for %s.%s' % (service, instance)
+        raise ApiFailure(error_message, 404)
+
+    min_instances = service_config.get_min_instances()
+
+    # Dump whatever number from the client to zk. get_instances() will limit
+    # readings from zk to [min_instances, max_instances].
     set_instances_for_marathon_service(service=service, instance=instance, instance_count=desired_instances)
-    response_body = {'desired_instances': desired_instances}
+    status = 'SUCCESS'
+    if desired_instances > max_instances:
+        desired_instances = max_instances
+        status = 'WARNING desired_instances is greater than max_instances %d' % max_instances
+    elif desired_instances < min_instances:
+        desired_instances = min_instances
+        status = 'WARNING desired_instances is less than min_instances %d' % min_instances
+
+    response_body = {'desired_instances': desired_instances, 'status': status}
     return Response(json_body=response_body, status_code=202)
```

Comment:
400 cannot be returned here because pyramid will take it as an exception and try to find the corresponding view handling. We either return 202 with warning or handle 400 error. Client would be surprised to see an error while the number of instance got changed to min_ or max_instances.